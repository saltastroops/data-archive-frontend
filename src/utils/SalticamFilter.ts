export type SalticamFilter =
  | "SalticamFilter_Name"
  | "340-35"
  | "380-40"
  | "B-S1"
  | "CLR-S1"
  | "Halpha-S1"
  | "Hbn-S1"
  | "Hbw-S1"
  | "I-S1"
  | "R-S1"
  | "Sb-S1"
  | "SDSSg-S1"
  | "SDSSi-S1"
  | "SDSSr-S1"
  | "SDSSu-S1"
  | "SDSSz-S1"
  | "SR613-21"
  | "SR708-25"
  | "SR815-29"
  | "SR862-32"
  | "Su-S1"
  | "Sv-S1"
  | "Sy-S1"
  | "U-S1"
  | "V-S1";

export interface ISalticamFilter {
  name: SalticamFilter;
  descriptiveName: string;
}
