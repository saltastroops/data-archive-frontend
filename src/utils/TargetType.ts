export type TargetType = string;

export const GALAXY = "10";

export const ISM = "20";

export const SOLAR_SYSTEM_BODY = "30";

export const STAR = "40";
