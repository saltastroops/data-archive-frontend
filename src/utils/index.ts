export const toTitleCase = (s: string) => {
  s = s.replace(/_/g, " ");
  return s.replace(/\w\S*/g, (txt: string) => {
    return txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase();
  });
};
