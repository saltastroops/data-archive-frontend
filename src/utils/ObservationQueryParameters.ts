// TypeScript interfaces describing the state of the search form

import { SalticamFilter } from "./SalticamFilter";
import { TargetType } from "./TargetType";

/**
 * An interface describing the React state of the observation query form.
 *
 * This state is used for populating all search form fields (and error messages)
 * and is parsed into the GraphQL query for searching observations.
 *
 * Properties:
 * -----------
 * general:
 *     General information about the observation.
 * target:
 *     Target details.
 * telescope:
 *     Telescope (and instrument) details.
 */
// TODO: Remove loading, replace with IObservationQueryParameters
export interface ISearchFormState {
  general: IGeneral;
  target: ITarget;
  telescope?: ITelescope;
  loading: boolean;
}

/**
 * The available calibration types.
 */
export type CalibrationType = "arc" | "bias" | "flat" | "standard";

/**
 * An interface for query parameters related to general information.
 *
 * Properties:
 * -----------
 * errors:
 *     Error messages.
 * observationNight:
 *     Date of the night when the observation was taken. This
 *     refers to the date whe the night started.
 * proposalCode:
 *     Identifier of the proposal to which the observation belongs.
 * proposalTitle:
 *     Title of the proposal to which the observation belongs.
 * calibrations:
 *     Calibration types to include in the search results
 *
 */
export interface IGeneralErrors {
  observationNight?: string;
  principalInvestigator?: string;
  proposalCode?: string;
  proposalTitle?: string;
  calibrations?: string;
}
export interface IGeneral {
  errors: IGeneralErrors;
  observationNight?: string;
  principalInvestigator?: string;
  proposalCode?: string;
  proposalTitle?: string;
  calibrations: Set<CalibrationType>;
}

/**
 * The available target name resolvers.
 */
export type TargetResolver = "Simbad" | "NED" | "VizieR";

/**
 * The available units for the search cone radius.
 */
export type SearchConeRadiusUnits = "arcseconds" | "arcminutes" | "degrees";

/**
 * An interface for query parameters related to target details.
 *
 * Properties:
 * -----------
 * declination:
 *     Declination.
 * errors:
 *     Errors.
 * name:
 *     Target name.
 * resolver:
 *     Resolver to use for resolving the target name to a position.
 * rightAscension:
 *     Right ascension.
 * searchConeRadius:
 *     Radius of the search cone radius, in the units given by
 *     searchConeRadiusUnits.
 * searchConeRadiusUnits:
 *     Units of the search cone radius given by searchConeRadius.
 *
 */
export interface ITarget {
  declination?: string;
  errors: ITargetErrors;
  loading?: string;
  name?: string;
  resolver: TargetResolver;
  rightAscension?: string;
  searchConeRadius?: string;
  searchConeRadiusUnits: SearchConeRadiusUnits;
  targetTypes: Set<TargetType>;
}

export interface ITargetErrors {
  name?: string;
  declination?: string;
  resolver?: string;
  rightAscension?: string;
  searchConeRadius?: string;
  searchConeRadiusUnits?: string;
  targetTypes?: string;
}

// TELESCOPES

/**
 * The supported telescope names.
 */
export type TelescopeName = "Lesedi" | "SALT" | "1.9 m";

/**
 * An interface for query parameters related to a telescope.
 *
 * This serves as a placeholder for more concrete interfaces such as ISALT.
 *
 * Properties:
 * -----------
 * name:
 *     Telescope name.
 */
export interface ITelescope {
  name?: TelescopeName;
}

/**
 * An interface for query parameters related to SALT.
 *
 * Properties:
 * -----------
 * instrument:
 *     A SALT instrument.
 * name:
 *     The string "SALT".
 */
export interface ISALT extends ITelescope {
  instrument: IInstrument;
  name: "SALT";
}

/**
 * An interface for query parameters related to Lesedi.
 *
 * Properties:
 * -----------
 * instrument:
 *     An instrument used on Lesedi.
 * name:
 *     The string "Lesedi".
 */
export interface ILesedi extends ITelescope {
  instrument: IInstrument;
  name: "Lesedi";
}

/**
 * An interface for query parameters related to the 1.9 m Telescope.
 *
 * Properties:
 * -----------
 * instrument:
 *     An instrument used on the 1./9 m Telescope.
 * name:
 *     The string "1.9 m".
 */
export interface IOneDotNineM extends ITelescope {
  instrument: IInstrument;
  name: "1.9 m";
}

// INSTRUMENTS

/**
 * The available instrument names.
 */
export type InstrumentName =
  | "Salticam"
  | "RSS"
  | "HRS"
  | "BVIT"
  | "SpUpNIC"
  | "SHOC"
  | "HIPPO";

/**
 * An interface for query parameters related to an instrument.
 *
 * This serves as a placeholder for more concrete interfaces such as ISalticam.
 *
 * Properties:
 * -----------
 * name:
 *     Telescope name.
 */
export interface IInstrument {
  name: InstrumentName;
}

/**
 * An interface for query parameters related to Salticam.
 *
 * Properties:
 * -----------
 * detectorMode:
 *     Detector mode.
 * errors:
 *     Errors.
 * exposureTime:
 *     Exposure time.
 * filter:
 *     Filter.
 * name:
 *     The string "Salticam".
 */
export interface ISalticam extends IInstrument {
  detectorMode?: "Normal" | "Slot Mode";
  errors: {
    detectorMode?: string;
    exposureTime?: string;
    filter?: string;
  };
  exposureTime?: string;
  filter?: SalticamFilter;
  name: "Salticam";
}

/**
 * An interface for query parameters related to RSS.
 *
 * Properties:
 * -----------
 * detectorMode:
 *     Detector mode.
 * errors:
 *     Errors.
 * exposureTime:
 *     Exposure time.
 * filter:
 *     Filter.
 * mode:
 *     Instrument mode specific details.
 * name:
 *     The string "RSS".
 */
export interface IRSS extends IInstrument {
  detectorMode?: RSSDetectorMode;
  errors: {
    detectorMode?: string;
    exposureTime?: string;
    filter?: string;
  };
  exposureTime?: string;
  filter?: string;
  modes?: IRSSModes;
  name: "RSS";
}

export interface IRSSModes {
  errors: {
    fabryPerotMode?: string;
    grating?: string;
    polarimetryModes?: string;
  };
  fabryPerotMode?: RSSFabryPerotMode;
  grating?: RSSGrating;
  names: Set<RSSInstrumentMode>;
  polarimetryModes?: Set<RSSPolarimetryMode>;
}

export type RSSDetectorMode = "Normal" | "Slot Mode" | "";

export type RSSGrating =
  | "Open"
  | "pg0300"
  | "pg0900"
  | "pg1300"
  | "pg1800"
  | "pg2300"
  | "pg3000"
  | "";

export type RSSInstrumentMode =
  | "Fabry Perot"
  | "FP polarimetry"
  | "Imaging"
  | "Polarimetric imaging"
  | "MOS"
  | "MOS polarimetry"
  | "Spectropolarimetry"
  | "Spectroscopy";

export type RSSFabryPerotMode = "HR" | "LR" | "MR" | "TF" | "";

export type RSSPolarimetryMode =
  | "All Stokes"
  | "Circular"
  | "Linear"
  | "Linear Hi"
  | "";

/**
 * An interface for query parameters related to HRS.
 *
 * Properties:
 * -----------
 * errors:
 *     Errors.
 * exposureTime:
 *     Exposure time.
 * mode:
 *     Resolution mode.
 * name:
 *     The string "HRS".
 */
export interface IHRS extends IInstrument {
  errors: {
    exposureTime?: string;
    mode?: string;
  };
  exposureTime?: string;
  mode?: HRSMode;
  name: "HRS";
}

export type HRSMode =
  | "High Resolution"
  | "High Stability"
  | "Int Cal Fibre"
  | "Low Resolution"
  | "Medium Resolution";

export interface IBVIT extends IInstrument {
  errors: {};
  filter?: BVITFilter;
  mode?: "Imaging" | "Streaming";
  name: "BVIT";
}

export type BVITFilter = "B" | "H-alpha" | "Open" | "R" | "U" | "V";

/**
 * An interface for query parameters related to HIPPO.
 *
 * Properties:
 * -----------
 * errors:
 *     Errors.
 * exposureTime:
 *     Exposure time.
 * filter:
 *     Filter.
 * name:
 *     The string "HIPPO".
 */
export interface IHIPPO extends IInstrument {
  errors: {
    exposureTime?: string;
    filter?: string;
  };
  exposureTime?: string;
  filter?: string;
  name: "HIPPO";
}

/**
 * An interface for query parameters related to SHOC.
 *
 * Properties:
 * -----------
 * errors:
 *     Errors.
 * exposureTime:
 *     Exposure time.
 * filter:
 *     Filter.
 * name:
 *     The string "SHOC".
 */
export interface ISHOC extends IInstrument {
  errors: {
    exposureTime?: string;
    filter?: string;
  };
  exposureTime?: string;
  filter?: string;
  name: "SHOC";
}

/**
 * An interface for query parameters related to SpUpNIC.
 *
 * Properties:
 * -----------
 * errors:
 *     Errors.
 * exposureTime:
 *     Exposure time.
 * filter:
 *     Filter.
 * name:
 *     The string "SpUpNIC".
 */
export interface ISpUpNIC extends IInstrument {
  errors: {
    exposureTime?: string;
    filter?: string;
  };
  exposureTime?: string;
  filter?: string;
  name: "SpUpNIC";
}

export interface IFile {
  id: string;
  filename: string;
  name: string;
  dataType: string;
  isReduced: boolean;
  targetName: string;
  rightAscension: number;
  declination: number;
  observationNight: string;
  category: string;
  telescope: string;
  instrument: string;
  observationId?: string;
  url?: string;
}
export interface IObservation {
  id: string;
  name: string | "Unknown Observation";
  proposal?: string;
  telescope: string;
  startTime: string;
  files: IFile[];
}
