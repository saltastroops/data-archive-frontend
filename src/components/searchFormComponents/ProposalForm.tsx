import * as React from "react";
import { IGeneral } from "../../utils/ObservationQueryParameters";
import { validateDate } from "../../utils/validators";
import { MainGrid, SubGrid } from "../basicComponents/Grids";
import InputField from "../basicComponents/InputField";

interface IProposalFormProps {
  proposal: IGeneral;
  onChange: (value: IGeneral) => void;
}

/**
 * A form for selecting general search parameters.
 */
class ProposalForm extends React.Component<IProposalFormProps, {}> {
  // Function for handling input changes.
  change = (e: React.ChangeEvent<HTMLInputElement>) => {
    const name = e.target.name;
    const value = e.target.value;
    this.props.onChange({
      ...this.props.proposal,
      [name]: value,
      errors: {
        ...this.props.proposal.errors,
        [name]: ""
      }
    });
  };

  render() {
    const {
      errors,
      principalInvestigator,
      observationNight,
      proposalCode,
      proposalTitle
    } = this.props.proposal;

    return (
      <>
        <MainGrid>
          <SubGrid>
            <p>Proposal code</p>
            <InputField
              data-test="proposal-code-input"
              error={errors.proposalCode}
              name={"proposalCode"}
              onChange={this.change}
              title={"The proposal code for SALT it looks like '2019-SCI-031'"}
              value={proposalCode || ""}
            />
          </SubGrid>
          <SubGrid>
            <p>Principal investigator</p>
            <InputField
              data-test="principal-investigator-input"
              name={"principalInvestigator"}
              value={principalInvestigator || ""}
              error={errors.principalInvestigator || ""}
              title={"Principal investigator for the proposal "}
              onChange={this.change}
            />
          </SubGrid>
        </MainGrid>

        <MainGrid>
          <SubGrid>
            <p>Proposal title</p>
            <InputField
              data-test="proposal-title-input"
              name={"proposalTitle"}
              value={proposalTitle || ""}
              error={errors.proposalTitle}
              title={"Program's title"}
              onChange={this.change}
            />
          </SubGrid>
          <SubGrid>
            <p>Observation night</p>
            <InputField
              data-test="observation-night-input"
              name={"observationNight"}
              value={observationNight || ""}
              error={errors.observationNight}
              title={`
                The night of the observation.\nYou can also input a date range by separating them with "..". E.g: 2019-01-01..2019-02-28
              `}
              onChange={this.change}
            />
          </SubGrid>
        </MainGrid>
      </>
    );
  }
}
export const validatedProposal = (general: IGeneral) => {
  return {
    ...general,
    errors: {
      observationNight: validateDate(general.observationNight || "")
    }
  };
};
export default ProposalForm;
