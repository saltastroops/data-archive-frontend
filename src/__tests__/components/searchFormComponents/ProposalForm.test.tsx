import { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import * as React from "react";
import ProposalForm from "../../../components/searchFormComponents/ProposalForm";
import { ProductType } from "../../../utils/ObservationQueryParameters";

const onChange = jest.fn();

describe("Proposal Form", () => {
  it("should render", () => {
    expect(
      <ProposalForm
        general={{ productTypes: new Set<ProductType>(), errors: {} }}
        onChange={onChange}
      />
    ).toBeDefined();
  });

  it("render correctly", () => {
    expect(
      toJson(
        shallow(
          <ProposalForm
            general={{ productTypes: new Set<ProductType>(), errors: {} }}
            onChange={onChange}
          />
        )
      )
    ).toMatchSnapshot();

    expect(
      toJson(
        shallow(
          <ProposalForm
            general={{
              errors: {},
              productTypes: new Set<ProductType>(),
              proposalTitle: "hello"
            }}
            onChange={onChange}
          />
        )
      )
    ).toMatchSnapshot();
    expect(
      toJson(
        shallow(
          <ProposalForm
            general={{
              errors: {
                principalInvestigator: "invalid Principal Investigator",
                proposalCode: "invalid proposal code",
                proposalTitle: "invalid proposal title"
              },
              productTypes: new Set<ProductType>(),
              proposalCode: "Code1"
            }}
            onChange={onChange}
          />
        )
      )
    ).toMatchSnapshot();
  });
});

describe("Proposal form on change", () => {
  it("should call onChange with the correct value", () => {
    const wrapper = mount(
      <ProposalForm
        general={{ productTypes: new Set<ProductType>(), errors: {} }}
        onChange={onChange}
      />
    );

    // Principal investigator

    const pi = wrapper.find('input[data-test="principal-investigator-input"]');
    pi.simulate("change", {
      target: { value: "John Doe", name: "principalInvestigator" }
    });
    expect(onChange).toBeCalledWith({
      errors: { principalInvestigator: "" },
      principalInvestigator: "John Doe",
      productTypes: new Set<ProductType>()
    });

    // Proposal code

    const proposalCode = wrapper.find('input[data-test="proposal-code-input"]');
    proposalCode.simulate("change", {
      target: { value: "2019-1-SCI-042", name: "proposalCode" }
    });
    expect(onChange).toBeCalledWith({
      errors: { proposalCode: "" },
      productTypes: new Set<ProductType>(),
      proposalCode: "2019-1-SCI-042"
    });

    // Proposal title

    const proposalTitle = wrapper.find(
      'input[data-test="proposal-title-input"]'
    );
    proposalTitle.simulate("change", {
      target: { value: "2019-1-SCI-042", name: "proposalTitle" }
    });
    expect(onChange).toBeCalledWith({
      errors: { proposalTitle: "" },
      productTypes: new Set<ProductType>(),
      proposalTitle: "2019-1-SCI-042"
    });
  });
});
