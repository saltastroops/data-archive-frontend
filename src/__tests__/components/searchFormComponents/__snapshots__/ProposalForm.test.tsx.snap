// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Proposal Form render correctly 1`] = `
<Fragment>
  <styled.div>
    <styled.div>
      <styled.div>
        <p>
          Proposal code
        </p>
        <HelpButton
          toolTipMessage="Proposal code, such as 2019-1-SCI-042."
        />
      </styled.div>
      <InputField
        data-test="proposal-code-input"
        name="proposalCode"
        onChange={[Function]}
        value=""
      />
    </styled.div>
    <styled.div>
      <styled.div>
        <p>
          Principal investigator
        </p>
        <HelpButton
          toolTipMessage="Principal investigator for the proposal."
        />
      </styled.div>
      <InputField
        data-test="principal-investigator-input"
        error=""
        name="principalInvestigator"
        onChange={[Function]}
        value=""
      />
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div>
      <styled.div>
        <p>
          Proposal title
        </p>
        <HelpButton
          toolTipMessage="Proposal title."
        />
      </styled.div>
      <InputField
        data-test="proposal-title-input"
        name="proposalTitle"
        onChange={[Function]}
        value=""
      />
    </styled.div>
    <styled.div>
      <styled.div>
        <p>
          Observation night
        </p>
        <HelpButton
          toolTipMessage="Night of the observation. You can input a date range by separating two dates with \\"...\\" For example: 2019-01-01 .. 2019-02-28."
        />
      </styled.div>
      <InputField
        data-test="observation-night-input"
        name="observationNight"
        onChange={[Function]}
        value=""
      />
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div>
      <h5
        className="title is-5"
      >
        Data category
      </h5>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div>
      <label>
        <styled.span>
          <styled.input
            checked={false}
            data-test="science-checkbox"
            id="science-checkbox"
            name="Science"
            onChange={[Function]}
          />
        </styled.span>
        <styled.span>
          Science
        </styled.span>
      </label>
      <label>
        <styled.span>
          <styled.input
            checked={false}
            data-test="arcs-checkbox"
            id="arcs-checkbox"
            name="Arc"
            onChange={[Function]}
          />
        </styled.span>
        <styled.span>
          Arcs
        </styled.span>
      </label>
      <label>
        <styled.span>
          <styled.input
            checked={false}
            data-test="biases-checkbox"
            id="biases-checkbox"
            name="Bias"
            onChange={[Function]}
          />
        </styled.span>
        <styled.span>
          Biases
        </styled.span>
      </label>
      <label>
        <styled.span>
          <styled.input
            checked={false}
            data-test="flats-checkbox"
            id="flats-checkbox"
            name="Flat"
            onChange={[Function]}
          />
        </styled.span>
        <styled.span>
          Flats
        </styled.span>
      </label>
      <label>
        <styled.span>
          <styled.input
            checked={false}
            data-test="standards-checkbox"
            id="standards-checkbox"
            name="Standard"
            onChange={[Function]}
          />
        </styled.span>
        <styled.span>
          Standard
        </styled.span>
      </label>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div>
      <h5
        className="title is-5"
      >
        Include rejected observations
      </h5>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div>
      <label>
        <styled.span>
          <styled.input
            checked={false}
            data-test="rejected-checkbox"
            id="rejected-checkbox"
            name="Rejected"
            onChange={[Function]}
          />
        </styled.span>
        <styled.span>
          Rejected
        </styled.span>
      </label>
    </styled.div>
  </styled.div>
</Fragment>
`;

exports[`Proposal Form render correctly 2`] = `
<Fragment>
  <styled.div>
    <styled.div>
      <styled.div>
        <p>
          Proposal code
        </p>
        <HelpButton
          toolTipMessage="Proposal code, such as 2019-1-SCI-042."
        />
      </styled.div>
      <InputField
        data-test="proposal-code-input"
        name="proposalCode"
        onChange={[Function]}
        value=""
      />
    </styled.div>
    <styled.div>
      <styled.div>
        <p>
          Principal investigator
        </p>
        <HelpButton
          toolTipMessage="Principal investigator for the proposal."
        />
      </styled.div>
      <InputField
        data-test="principal-investigator-input"
        error=""
        name="principalInvestigator"
        onChange={[Function]}
        value=""
      />
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div>
      <styled.div>
        <p>
          Proposal title
        </p>
        <HelpButton
          toolTipMessage="Proposal title."
        />
      </styled.div>
      <InputField
        data-test="proposal-title-input"
        name="proposalTitle"
        onChange={[Function]}
        value="hello"
      />
    </styled.div>
    <styled.div>
      <styled.div>
        <p>
          Observation night
        </p>
        <HelpButton
          toolTipMessage="Night of the observation. You can input a date range by separating two dates with \\"...\\" For example: 2019-01-01 .. 2019-02-28."
        />
      </styled.div>
      <InputField
        data-test="observation-night-input"
        name="observationNight"
        onChange={[Function]}
        value=""
      />
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div>
      <h5
        className="title is-5"
      >
        Data category
      </h5>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div>
      <label>
        <styled.span>
          <styled.input
            checked={false}
            data-test="science-checkbox"
            id="science-checkbox"
            name="Science"
            onChange={[Function]}
          />
        </styled.span>
        <styled.span>
          Science
        </styled.span>
      </label>
      <label>
        <styled.span>
          <styled.input
            checked={false}
            data-test="arcs-checkbox"
            id="arcs-checkbox"
            name="Arc"
            onChange={[Function]}
          />
        </styled.span>
        <styled.span>
          Arcs
        </styled.span>
      </label>
      <label>
        <styled.span>
          <styled.input
            checked={false}
            data-test="biases-checkbox"
            id="biases-checkbox"
            name="Bias"
            onChange={[Function]}
          />
        </styled.span>
        <styled.span>
          Biases
        </styled.span>
      </label>
      <label>
        <styled.span>
          <styled.input
            checked={false}
            data-test="flats-checkbox"
            id="flats-checkbox"
            name="Flat"
            onChange={[Function]}
          />
        </styled.span>
        <styled.span>
          Flats
        </styled.span>
      </label>
      <label>
        <styled.span>
          <styled.input
            checked={false}
            data-test="standards-checkbox"
            id="standards-checkbox"
            name="Standard"
            onChange={[Function]}
          />
        </styled.span>
        <styled.span>
          Standard
        </styled.span>
      </label>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div>
      <h5
        className="title is-5"
      >
        Include rejected observations
      </h5>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div>
      <label>
        <styled.span>
          <styled.input
            checked={false}
            data-test="rejected-checkbox"
            id="rejected-checkbox"
            name="Rejected"
            onChange={[Function]}
          />
        </styled.span>
        <styled.span>
          Rejected
        </styled.span>
      </label>
    </styled.div>
  </styled.div>
</Fragment>
`;

exports[`Proposal Form render correctly 3`] = `
<Fragment>
  <styled.div>
    <styled.div>
      <styled.div>
        <p>
          Proposal code
        </p>
        <HelpButton
          toolTipMessage="Proposal code, such as 2019-1-SCI-042."
        />
      </styled.div>
      <InputField
        data-test="proposal-code-input"
        error="invalid proposal code"
        name="proposalCode"
        onChange={[Function]}
        value="Code1"
      />
    </styled.div>
    <styled.div>
      <styled.div>
        <p>
          Principal investigator
        </p>
        <HelpButton
          toolTipMessage="Principal investigator for the proposal."
        />
      </styled.div>
      <InputField
        data-test="principal-investigator-input"
        error="invalid Principal Investigator"
        name="principalInvestigator"
        onChange={[Function]}
        value=""
      />
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div>
      <styled.div>
        <p>
          Proposal title
        </p>
        <HelpButton
          toolTipMessage="Proposal title."
        />
      </styled.div>
      <InputField
        data-test="proposal-title-input"
        error="invalid proposal title"
        name="proposalTitle"
        onChange={[Function]}
        value=""
      />
    </styled.div>
    <styled.div>
      <styled.div>
        <p>
          Observation night
        </p>
        <HelpButton
          toolTipMessage="Night of the observation. You can input a date range by separating two dates with \\"...\\" For example: 2019-01-01 .. 2019-02-28."
        />
      </styled.div>
      <InputField
        data-test="observation-night-input"
        name="observationNight"
        onChange={[Function]}
        value=""
      />
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div>
      <h5
        className="title is-5"
      >
        Data category
      </h5>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div>
      <label>
        <styled.span>
          <styled.input
            checked={false}
            data-test="science-checkbox"
            id="science-checkbox"
            name="Science"
            onChange={[Function]}
          />
        </styled.span>
        <styled.span>
          Science
        </styled.span>
      </label>
      <label>
        <styled.span>
          <styled.input
            checked={false}
            data-test="arcs-checkbox"
            id="arcs-checkbox"
            name="Arc"
            onChange={[Function]}
          />
        </styled.span>
        <styled.span>
          Arcs
        </styled.span>
      </label>
      <label>
        <styled.span>
          <styled.input
            checked={false}
            data-test="biases-checkbox"
            id="biases-checkbox"
            name="Bias"
            onChange={[Function]}
          />
        </styled.span>
        <styled.span>
          Biases
        </styled.span>
      </label>
      <label>
        <styled.span>
          <styled.input
            checked={false}
            data-test="flats-checkbox"
            id="flats-checkbox"
            name="Flat"
            onChange={[Function]}
          />
        </styled.span>
        <styled.span>
          Flats
        </styled.span>
      </label>
      <label>
        <styled.span>
          <styled.input
            checked={false}
            data-test="standards-checkbox"
            id="standards-checkbox"
            name="Standard"
            onChange={[Function]}
          />
        </styled.span>
        <styled.span>
          Standard
        </styled.span>
      </label>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div>
      <h5
        className="title is-5"
      >
        Include rejected observations
      </h5>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div>
      <label>
        <styled.span>
          <styled.input
            checked={false}
            data-test="rejected-checkbox"
            id="rejected-checkbox"
            name="Rejected"
            onChange={[Function]}
          />
        </styled.span>
        <styled.span>
          Rejected
        </styled.span>
      </label>
    </styled.div>
  </styled.div>
</Fragment>
`;
